### PR TITLE
Add illumos support and update dependencies for newest Constants in sys/unix

### PR DESCRIPTION
**Pull request by Toasterson**
This adds support for illumos, based on tcell work by gdamore and the tcell authors.

Original work by @cneira for the nomad zone driver.

### CODE REVIEW

**Comment by dims** [comment]
LGTM

File: go.mod [hunk 1]
```diff
@@ -1,12 +1,14 @@
+// This is a generated file. Do not edit directly.
```
Comment (thaJeztah) [inline go.mod:1]: Looks like this is not correct?

Comment (Toasterson) [reply go.mod]: Oh yes. this snuck in. Fixed

**Comment by Toasterson** [comment]
Squashed my commits and cleaned the incorrect author tag.

**Pull request merged**